specification/api-correlationcontext.md

```diff
@@ -149,37 +152,83 @@ Enter Scope 1
+provides a function which takes a context and a key as input, and returns a
+value. Returns the `EntryValue` associated with the given `EntryKey`, or null
+if the given `EntryKey` is not present.
 
 Required parameter:
 
+`Context` the context from which to get the correlation.
+
 `EntryKey` entry key to return the value for.
 
+### SetCorrelation
+
+To record the value for a label, the Correlations API provides a function which
+takes a context, a key, and a value as input. Returns an updated `Context` which
+contains the new value.
+
+Required parameter:
+
+`Context` the context to which the value will be set.
+
+`EntryKey` entry key for which to set the value.
+
+`EntryValue` entry value to set.
+
+### RemoveCorrelation
+
+To delete a label, the Correlations API provides a function which takes a context
+and a key as input. Returns an updated `Context` which no longer contains the selected `EntryKey`.
+
+Required parameter:
+
+`Context` the context from which to remove the correlation.
+
+`EntryKey` entry key to remove.
+
+### ClearCorrelations
+
+To avoid sending any labels to an untrusted process, the Correlation API provides
+a function to remove all Correlations from a context. Returns an updated `Context`
+with no correlations.
+
+Required parameter:
+
+`Context` the context from which to remove all correlations.
+
+### GetCorrelationPropagator
+
+To deserialize the previous labels set by prior processes, and to serialize the current
+total set of labels and send them to the next process, the Correlations API provides a function
+which returns a Correlation-specific implementation of the `HTTPExtract` and `HTTPInject`
+functions found in the Propagation API.
+
 ### Limits
 
 Combined size of all entries should not exceed 8192 bytes before encoding.
 The size restriction applies to the deserialized entries so that the set of decoded
- `DistributedContext`s is independent of the encoding format.
+ `CorrelationContext`s is independent of the encoding format.
 
-### DistributedContext Propagation
+### CorrelationContext Propagation
 
-`DistributedContext` may be propagated across process boundaries or across any arbitrary boundaries
+`CorrelationContext` may be propagated across process boundaries or across any arbitrary boundaries
 (process, $OTHER_BOUNDARY1, $OTHER_BOUNDARY2, etc) for various reasons.
 For example, one may propagate 'project-id' Entry across all micro-services to break down metrics
-by 'project-id'. Not all entries in a `DistributedContext` should be propagated and not all entries in a `DistributedContext`
-should be accepted from a remote peer. Hence, `DistributedContext` propagator must allow specifying an optional
+by 'project-id'. Not all entries in a `CorrelationContext` should be propagated and not all entries in a `CorrelationContext`
+should be accepted from a remote peer. Hence, `CorrelationContext` propagator must allow specifying an optional
```

Comment:
I don't think this stuff belongs in the **API specification**. Happy to see tuff like this in the SDK specification. The SDK will provide propagators for W3C correlation context. The propagators can apply filtering if they wish.

Reply:
Removed